### DISCUSSION TITLE
Blank Screen Problems with SQL Plugins

### DISCUSSION THREAD

**Discussion by ppachan** [coder/code-server#3969] [Q&A]
I installed many of the MySQL plugins on Code Server - but all of them give problem with the display. The initial display page (to setup connection to database) page appears blank - so not able to use any of these plugins. 

The plugins I tried are: 
1) https://marketplace.visualstudio.com/items?itemName=cweijan.vscode-mysql-client2 
2) https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools-driver-mysql
 3) https://marketplace.visualstudio.com/items?itemName=vs-publisher-988541.apexsql-power-tools

Since all of these tools give the same problem, I am thinking, there is something wrong with code-server or something wrong that I am doing...
I installed these using --install-extension option.

One of them (#2) worked on a Mac Machine; but not working on an Ubuntu machine (same blank screen problem)

**Comment by jsjoeio** [comment]
Hmm...So technically installing extensions from the Marketplace in code-server is against their ToS so I can't use those, but if they are in [Open VSX](https://open-vsx.org/), I can try reproducing. Feel free to put repro steps and I'm happy to try on my local Mac.

**Reply by ppachan** [reply]
Thanks jsjoeio. 

Here we go:

1) Downloaded from VSIX: https://open-vsx.org/extension/cweijan/vscode-mysql-client2

2) Installed on my ubuntu machine:
$code-server/release>node .  --install-extension ~/vsix/cweijan.vscode-mysql-client2-3.9.3.vsix
3) Start code-server

4) Click the icon for the plugin (database icon), Click + sign to open a new database connection . A new tab opens, titled "connect". But it is entirely blank! See attached. 

<img width="1425" alt="Screenshot 2021-08-13 at 8 53 53 AM" src="https://user-images.githubusercontent.com/32189861/129301109-522cfbe5-3936-4fcd-818c-6eba760dd103.png">

**Reply by ppachan** [reply]
To isolate any issues with the specific plugin, trying a different one. 

1) Download https://open-vsx.org/extension/mtxr/sqltools and its database specific driver plugin: https://open-vsx.org/extension/mtxr/sqltools-driver-mysql
2) Install both of them, using --install-extension
3) Start code-server
4) Click on plugin's icon. Click "Start New Connection". A new tab opens, but it is completely blank! Once again!

<img width="1435" alt="Screenshot 2021-08-13 at 9 07 14 AM" src="https://user-images.githubusercontent.com/32189861/129301878-54fedaa1-fb90-43c1-bc2f-ffd395a3ee8b.png">

**Reply by jsjoeio** [reply]
Thank you for those! Super easy to follow. Strangely enough, I get the blank page, but then when I clear it and refresh, it shows what it's supposed to show. I'm not seeing any errors in the console or in the terminal 🤔 Let me know if refreshing works for you. 

## Video
Here's a video of me doing this:

https://user-images.githubusercontent.com/3806031/129391656-205b712a-e4c4-4b62-b155-93c39250c70f.mov

**Comment by ppachan** [comment]
I tried everything... still not able to get it working... recently tried on the new version of code server...

**Reply by jsjoeio** [reply]
Are you accessing code-server via HTTPS?

Are there any logs in the browser?
